core/src/util/AndroidCompat.h

```diff
@@ -0,0 +1,56 @@
+/*=============================================================================
+
+ Library: CppMicroServices
+
+ Copyright (c) The CppMicroServices developers. See the COPYRIGHT
+ file at the top-level directory of this distribution and at
+ https://github.com/saschazelzer/CppMicroServices/COPYRIGHT .
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+
+ =============================================================================*/
+
+#ifndef ANDROIDCOMPAT_H
+#define ANDROIDCOMPAT_H
+
+/**
+ * Compatibility layer for Android
+ *
+ * This file and AndroidCompat.cpp exist to allow CppMicroServices
+ * to use std::to_string function without breaking the build on Android.
+ * Once Android starts supporting std::to_string function, this file and
+ * AndroidCompat.cpp files can be discarded.
+ */
+
+#if defined(__ANDROID__)
+
+#include <string>
+
+namespace detail_AndroidCompat {
+std::string to_string(int val);
+std::string to_string(unsigned val);
+std::string to_string(long val);
+std::string to_string(unsigned long val);
+std::string to_string(long long val);
+std::string to_string(unsigned long long val);
+std::string to_string(float val);
+std::string to_string(double val);
+std::string to_string(long double val);
+}
+
+namespace std {
+using namespace detail_AndroidCompat;
```

Comment:
We need to include "some" header containing the `us::ToString` function 
declaration (I suggest to use our camel case style, makes it also easier 
to search for the std variant). If we can re-use an existing one, like 
usUtil_p.h, that would be fine. My point is, that it would be nice to 
confine the android specifics to the function definition in the 
according cpp file.


On 07/27/2016 07:37 PM, kevinleeMW wrote:
>
> In core/src/util/AndroidCompat.h 
> <https://github.com/CppMicroServices/CppMicroServices/pull/111#discussion_r72485782>:
>
> > +#include <string>
> > +
> > +namespace detail_AndroidCompat {
> > +std::string to_string(int val);
> > +std::string to_string(unsigned val);
> > +std::string to_string(long val);
> > +std::string to_string(unsigned long val);
> > +std::string to_string(long long val);
> > +std::string to_string(unsigned long long val);
> > +std::string to_string(float val);
> > +std::string to_string(double val);
> > +std::string to_string(long double val);
> > +}
> > +
> > +namespace std {
> > +using namespace detail_AndroidCompat;
>
> Sascha,
>
> I just wanted to clarify: does your suggestion entail having to 
> include an AndroidCompat header (that contains |us::to_string()|) 
> everywhere we need to use |to_string()|?
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub 
> <https://github.com/CppMicroServices/CppMicroServices/pull/111/files/7f535382327a6260b737f32961011600b7734320#r72485782>, 
> or mute the thread 
> <https://github.com/notifications/unsubscribe-auth/AANXpS7yEloJHX7wiT4o01ed_yBIx8LLks5qZ5c-gaJpZM4JVgwJ>.
>